src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -31,6 +33,11 @@ internal sealed class ExceptionProcessor : BaseProcessor<Activity>
 
     public ExceptionProcessor()
     {
+#if NET6_0_OR_GREATER
```

Comment:
@vitek-karas Could we use the public API for net462 as well?

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getexceptionpointers?view=netframework-4.6.2